constraint-solver/src/range_constraint.rs

```diff
@@ -103,6 +104,59 @@ impl<T: FieldElement> RangeConstraint<T> {
         in_range && in_mask
     }
 
+    /// Returns `Some(x)` if `x` is the only value in the set of values
+    /// allowed by the range constraint such that (in the integers)
+    /// `x = offset (mod modulus)`.
+    /// In other words, there is a unique integer `k` and a unique
+    /// integer `x` in the allowed values seen as integers such that
+    /// `offset = k * modulus + x`.
+    pub fn has_unique_modular_solution(&self, offset: T, modulus: T) -> Option<T> {
+        // println!(
+        //     "has_unique_modular_solution on self = {self}: offset = {}, modulus = {}",
+        //     offset, modulus
+        // );
+        // If the modulus is larger than half the field, the mapping to integers
+        // is not obvious. Also, if the number of values in the range constraint
+        // is at least two times the modulus, there are always at least to solutions.
+        if !modulus.is_in_lower_half() {
```

Comment:
I think this should be an assertion and the transformation should be pulled up into the caller.